sequence.go

```diff
@@ -0,0 +1,39 @@
+package ksuid
+
+import (
+	"encoding/binary"
+	"errors"
+	"math"
+)
+
+// Sequence is a KSUID generator which produces a sequence of ordered KSUIDs
+// from a seed.
+//
+// Up to 65536 KSUIDs can be generated by for a single seed.
+//
+// A typical usage of a Sequence looks like this:
+//
+//	seq := ksuid.Sequence{
+//		Seed: ksuid.New(),
+//	}
+//	id, err := seq.Next()
+//
+type Sequence struct {
```

Comment:
Should we document the implementation doesn't guarantee thread safety?

Reply:
Sure, I'll add that.